### ISSUE TITLE
[FEATURE] Support multi-output for `TrainOnlyTransformerMixin`

### ISSUE THREAD

**Issue by CarloLepelaars** [koaning/scikit-lego#699]
`TrainOnlyTransformerMixin.fit` checks if `y` is 1D. This breaks if you are training multi-label models where `y` is 2D.

To fix this we only need to add `multi_output=True` to `check_X_y(X, y, estimator=self)`.
Line in question:
https://github.com/koaning/scikit-lego/blob/236f4919caf5da736ee5850aabc90850054ae525/sklego/common.py#L84

The only added thing is that `y` can't contain `np.nan` or `np.inf` anymore, which I think is a good thing.

`check_X_y` docs:
https://scikit-learn.org/stable/modules/generated/sklearn.utils.check_X_y.html

I will implement this and make sure all tests still pass.

**Comment by FBruzzesi** [comment]
Closed by #700
